### PR TITLE
Update main readme and add supporting docs

### LINKED ISSUE #132: Document how to contribute to the development of the plugin
As @jffng [requested here](https://github.com/WordPress/create-block-theme/pull/126#pullrequestreview-1156585188) I guess we should add some documentation on how to contribute with code to the plugin.

### PR DESCRIPTION
**Pull request by mikachan**
This PR updates the main readme and adds some supporting docs, inspired mainly by the Gutenberg repo.

- Adds further info to the main readme
- Adds a Contributing file
- Adds a Code of Conduct file

I'm hoping this will help welcome and introduce new users and contributors to the plugin!

Addresses #132.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,57 +1,84 @@
...
+
+It works alongside features that are already available in the Site Editor to enhance the workflow for creating block themes. After being tested in this plugin, some of the features included here may be moved into the Site Editor itself.
 
 This plugin allows you to:
-- Create a new theme, blank theme, child theme or style variation.
-- Embed Google Fonts in your theme
-- Embed local font assets in your theme
 
-The plugin is development only — not intended for use on production websites, but used as a tool to create new themes.
+-   Create a new theme, blank theme, child theme, or style variation from within the Site Editor
```
Comment (jffng) [inline README.md]:
```suggestion
- Export your existing theme with all customizations made in the Site Editor
- Create a new theme, blank theme, child theme, or style variation from the Site Editor
```

File: README.md [hunk 2]
```diff
@@ -1,57 +1,84 @@
...
+-   Embed Google Fonts locally in your theme
+-   Embed local font assets in your theme
+
+Learn more about [how to use the plugin](#how-to-use-the-plugin) or [how to contribute](#how-to-contribute).
+
+_The plugin is for development only — it is not intended for use on production websites, but to be used as a tool to create new themes._
```
Comment (jffng) [inline README.md]: While we're here, what do you think we should move this line higher, perhaps below line 5?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+-   [Node.js](https://nodejs.org/en/) (v16.9.1)
+-   We recommend using [Node Version Manager](https://github.com/nvm-sh/nvm) (nvm) to manage your Node.js versions
+-   WordPress Development Site, such as [wp-env](https://github.com/WordPress/gutenberg/blob/trunk/packages/env/README.md) or [Local](https://localwp.com/)
+
+We recommend following the [Gutenberg code contribution guide](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/getting-started-with-code-contribution.md) for more details on setting up a development environment.
+
+#### Code Setup
+
+Fork the Create Block Theme repository, clone it to your computer and add the WordPress repository as upstream.
```
Comment (jffng) [inline CONTRIBUTING.md]:
```suggestion
[Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the Create Block Theme repository, [clone it to your computer](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) and add the WordPress repository as [upstream](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-repository-for-a-fork)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+$ git clone https://github.com/YOUR_GITHUB_USERNAME/create-block-theme.git
+$ cd create-block-theme
+$ git remote add upstream https://github.com/WordPress/create-block-theme.git
+```
+
+Run the following commands to install the plugin dependencies:
+
+```
+npm install
+composer install
```
Comment (jffng) [inline CONTRIBUTING.md:40]: This would require composer to be installed globally — should we add that to the pre-reqs?

**Review by jffng — COMMENTED** [review]
Looking good, thanks so much for adding this.

File: README.md (hunk 2 above)
Comment (mikachan) [reply README.md]: Yep, sounds good! Done in https://github.com/WordPress/create-block-theme/pull/220/commits/875db3dee1da0573fcac3c78575e1aebe0714c08.

File: CONTRIBUTING.md (hunk 4 above)
Comment (mikachan) [reply CONTRIBUTING.md:40]: Right, I think we should. I've updated the pre-reqs here: https://github.com/WordPress/create-block-theme/pull/220/commits/dd049d330287e48fec6c0b8a7c9252fa13f81dd9

**Review by jffng — APPROVED** [review]
🚢

**Pull request merged**